### ISSUE TITLE
Not NAR mode in training

### ISSUE THREAD

**Issue by Yongxin-Zhu** [Rongjiehuang/TranSpeech#3]
The training code still use the causal attention mask. You need to set full_context_alignment=True in the decoder.forward function to turn non-causal attention mask. Is it a mistake?

**Comment by steventan0110** [comment]
### Causal Mask used in reported results?
same question here. I'm benchmarking model performance with and without the causal mask and will put the result up (very soon I hope). Just want to confirm with the author @Rongjiehuang if such a causal mask is used for results reported in the paper? (I don't think it's necessarily a mistake but with causal masks, it does affect the decoder's capacity)

###  Some questions about the benchmark
I'm wondering if you (@Rongjiehuang) could provide more details about how you benchmark facebook's S2UT paper (Lee et al.,) on the CVSS-C dataset? I use fairseq's open-sourced code to replicate their paper but obtain a very different result compared to the numbers reported in your paper:
I got BLEU=**18.2** for en->es translation and BLEU=**20.9** for en->fr translation, while in your paper the numbers are **10.07** for en->es and **15.28** for en->fr.  I also had preliminary results for replicating your experiments (which I will put here after I benchmark btw with/without causal mask) and the results are also higher than your paper's number. Therefore, I believe there must be some setup difference and could you check my setup below to see if anything could go wrong?

I first used facebook's released mHubert model to cluster units for the es/fr data. Then I train transformer-based S2UT (following Lee et al.,) on the **reduced** units as their paper showed that directly working with deduplicated reduced-units obtain better performance. Then the bleu score is also computed using hyp/ref that are reduced-units. I also replicate your experiments with target units that are already deduplicated.

Lastly, thanks for open-sourcing the code! Hope to hear from you soon.

P.S. @youngsheen  I saw you are the first author of DiffS2UT and I'm wondering if the code for that paper is released anywhere? I'm very interested in your approach as well.
